src/db/z_db.erl

```diff
@@ -847,6 +861,7 @@ merge_props([R|Rest], Acc) ->
     end.
 
 
+-spec assoc1(atom(), #context{}, string(), [tuple()], pos_integer()) -> {ok, [[tuple()]]}.
```

Comment:
The spec in line 864 uses a record, please define a type for the record and use that instead.